samples/extensions/sparse_image/sparse_image.cpp

```diff
@@ -0,0 +1,1644 @@
+ */
+void SparseImage::update_mvp()
+{
+	// Update MVP + new mip block calculations
+	MVP mvp_ubo   = {};
+	mvp_ubo.model = glm::mat4(1.0f);
+	mvp_ubo.view  = camera.matrices.view;
+	mvp_ubo.proj  = camera.matrices.perspective;
+
+	mvp_buffer->update(&mvp_ubo, sizeof(mvp_ubo));
+
+	current_mvp_transform = mvp_ubo.proj * mvp_ubo.view * mvp_ubo.model;
+}
+
+void SparseImage::build_command_buffers()
+{
+	VkCommandBufferBeginInfo command_buffer_begin_info = vkb::initializers::command_buffer_begin_info();
+
+	// Clear color and depth values.
+	VkClearValue clear_values[2];
+	clear_values[0].color        = {{0.0f, 0.0f, 0.0f, 0.0f}};
+	clear_values[1].depthStencil = {0.0f, 0};
+
+	// Begin the render pass.
+	VkRenderPassBeginInfo render_pass_begin_info    = vkb::initializers::render_pass_begin_info();
+	render_pass_begin_info.renderPass               = render_pass;
+	render_pass_begin_info.renderArea.offset.x      = 0;
+	render_pass_begin_info.renderArea.offset.y      = 0;
+	render_pass_begin_info.renderArea.extent.width  = width;
+	render_pass_begin_info.renderArea.extent.height = height;
+	render_pass_begin_info.clearValueCount          = 2U;
+	render_pass_begin_info.pClearValues             = clear_values;
+
+	for (size_t i = 0U; i < draw_cmd_buffers.size(); ++i)
+	{
+		auto cmd = draw_cmd_buffers[i];
+
+		// Begin command buffer.
+		vkBeginCommandBuffer(cmd, &command_buffer_begin_info);
+
+		// Set framebuffer for this command buffer.
+		render_pass_begin_info.framebuffer = framebuffers[i];
+		// We will add draw commands in the same command buffer.
+		vkCmdBeginRenderPass(cmd, &render_pass_begin_info, VK_SUBPASS_CONTENTS_INLINE);
+
+		// Bind the graphics pipeline.
+		vkCmdBindPipeline(cmd, VK_PIPELINE_BIND_POINT_GRAPHICS, sample_pipeline);
+
+		// Set viewport dynamically
+		VkViewport viewport = vkb::initializers::viewport(static_cast<float>(width), static_cast<float>(height), 0.0f, 1.0f);
+		vkCmdSetViewport(cmd, 0U, 1U, &viewport);
+
+		// Set scissor dynamically
+		VkRect2D scissor = vkb::initializers::rect2D(width, height, 0, 0);
+		vkCmdSetScissor(cmd, 0U, 1U, &scissor);
+
+		vkCmdBindDescriptorSets(draw_cmd_buffers[i], VK_PIPELINE_BIND_POINT_GRAPHICS, sample_pipeline_layout, 0U, 1U, &descriptor_set, 0U, NULL);
+
+		VkDeviceSize offsets[1] = {0};
+		vkCmdBindVertexBuffers(draw_cmd_buffers[i], 0U, 1U, &vertex_buffer->get_handle(), offsets);
+		vkCmdBindIndexBuffer(draw_cmd_buffers[i], index_buffer->get_handle(), 0U, VK_INDEX_TYPE_UINT16);
+		vkCmdDrawIndexed(draw_cmd_buffers[i], index_count, 1U, 0U, 0U, 0U);
+
+		// Draw user interface.
+		draw_ui(draw_cmd_buffers[i]);
+
+		// Complete render pass.
+		vkCmdEndRenderPass(cmd);
+
+		// Complete the command buffer.
+		VK_CHECK(vkEndCommandBuffer(cmd));
+	}
+}
+
+void SparseImage::process_texture_blocks()
+{
+	uint8_t block_counter;
+	block_counter = std::min<size_t>(blocks_to_update_per_cycle, virtual_texture.texture_block_update_set.size());
```

Comment:
With both arguments to `std::min` are of the same type, you don't need to specify that type as the template argument.
That is: `std::min<size_t>` can be reduced to just `std::min` again.